File: greedy.c
 #include <stdio.h>
 #include "cs50.h"
 #include <math.h>

 int main(void)
 {
     float n;
     int c = 0;
     int t;

          do
     {
         printf("How much change is owed?\n");
         n = get_float();
     }
     while (n < 0);
     n *= 100.0;
     t = round (n);
     
     while (t >= 25)
     {
         c++;
         t -= 25;
     }

     while (t >= 10)
     {
         c++;
         t -= 10;
     }

     while (t >= 5)
     {
         c++;
         t -= 5;
     }

     while (t >= 1)
     {
         c++;
         t -= 1;
     }
     printf("%d\n", c);

     return 0;
 }

File: pset6/greedy.py
import cs50

def main():
    print ("Hello, how much change is owed?", end="")
    print()
    change1 = cs50.get_float()

    coins = 0
    change = change1 * 100

    while change > 0:
        if change >= 25:
            change -= 25
            coins += 1

        elif change >= 10:
            change -= 10
            coins += 1

        elif change >= 5:
            change -= 5
            coins += 1

        elif change >= 1:
            change -= 1
            coins += 1

    print (coins)

if __name__ == "__main__":
    main()